include/klee/Core/TerminationTypes.h

```diff
@@ -0,0 +1,56 @@
+//===-- TerminationTypes.h --------------------------------------*- C++ -*-===//
+//
+//                     The KLEE Symbolic Virtual Machine
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef KLEE_TERMINATIONTYPES_H
+#define KLEE_TERMINATIONTYPES_H
+
+#include <cstdint>
+
+
+#define TERMINATION_TYPES \
+  TTYPE(RUNNING,0U,"") \
```

Comment:
Thanks for having a look!

> Also, why 0U and not just 0?
Because I want unsigned literals.

Reply:
But I don't see why here you need to be explicit with the "U" suffix.  It looks ugly, unnecessarily.

Reply:
Maybe not here but in the future one might re-use the macros in a different context and I'd prefer to assign the correct type to the literal instead of relying on implicit conversions (which cause too many bugs).

Reply:
Can I ask why you force unsigned to begin with?   All the code that I've seen uses ints, and avoids the ugly "1U".  I'd be curious to understand your choice.

Reply:
> Can I ask why you force unsigned to begin with?

Because this is intended domain and I store it in a 1 byte unsigned enum directly below.

> All the code that I've seen uses ints

For what? E.g. Misra C/C++ enforces 'U' for unsigned constants. For literals >2^31 C (imho) doesn't even specify what you actually get (signed or unsigned) and for smaller values you have to rely on the word width of the host. Too many rules to remember for me...

Reply:
enum class'es are by default backed by signed ints in C++11.   I feel you're over-optimising here -- as far as space goes, we won't have more than 100 termination reasons.  But as I said, if you really like this, ok, it's minor.

Reply:
> we won't have more than 100 termination reasons

But we might have a lot of leaf nodes with termination reason in svg files or sqlite databases.

Reply:
And if you stick with the default `int` we will waste a lot of space? ;)

Reply:
Ah, I see what you mean, but I don't understand what this has to do with signed vs unsigned, we will not need so many termination reasons.

Reply:
Anyway, you seem to really like this optimization, so we can keep it, although as a reader I was wondering why we need the explicit unsigned constants, when we have so few different termination categories.